src/generators/genlua.ml

```diff
@@ -1265,11 +1279,20 @@ and gen_tbinop ctx op e1 e2 =
 	    gen_expr ctx e2;
 	    spr ctx ")";
     | Ast.OpAdd,_,_ when (is_string_expr e1 || is_string_expr e2) ->
+            add_feature ctx "use._hx_string_add";
+            spr ctx "_hx_string_add(";
 	    gen_value ctx e1;
-	    print ctx " .. ";
+	    print ctx ",";
 	    gen_value ctx e2;
+            spr ctx ")";
```

Comment:
this _hx_string_add function will wrap concatenation in _hx_string_add(x,y), and make sure Std.string is called on the arguments.